src/backend/sgx/mod.rs

```diff
@@ -186,8 +187,9 @@ impl super::Thread for Thread {
         //
         //   4. Asynchronous exits other than invalid opcode will panic.
         loop {
-            how = match self.thread.enter(how, &mut self.block) {
-                Err(Some(ei)) if ei.trap == Exception::InvalidOpcode => Entry::Enter,
+            registers.rdi = (&mut self.block).into();
```

Comment:
```suggestion
            registers.rdx = (&mut self.block).into();
```

Reply:
With this, all works again :+1: good job!

Reply:
Thank you! I was banging my head against a wall on this one...